File: backups/consolidation_20250909_033925/Backend/Python/api/captain_api.py
"""
Captain Tools API

This module provides a Flask API for the captain tools,
allowing them to be called from the NodeJS server.
"""

from flask import Flask, jsonify
from captain_tools import (
    captain_score_predictor,
    vice_captain_optimizer,
    loophole_detector,
    form_based_captain_analyzer,
    matchup_based_captain_advisor
)

app = Flask(__name__)

@app.route('/api/captain/score-predictor', methods=['GET'])
def api_captain_score_predictor():
    """API endpoint for Captain Score Predictor"""
    try:
        results = captain_score_predictor()
        return jsonify({"status": "ok", "players": results})
    except Exception as e:
        return jsonify({"status": "error", "message": str(e)})

@app.route('/api/captain/vice-captain-optimizer', methods=['GET'])
def api_vice_captain_optimizer():
    """API endpoint for Vice-Captain Optimizer"""
    try:
        results = vice_captain_optimizer()
        return jsonify({"status": "ok", "combinations": results})
    except Exception as e:
        return jsonify({"status": "error", "message": str(e)})

@app.route('/api/captain/loophole-detector', methods=['GET'])
def api_loophole_detector():
    """API endpoint for Loophole Detector"""
    try:
        results = loophole_detector()
        return jsonify({"status": "ok", "data": results})
    except Exception as e:
        return jsonify({"status": "error", "message": str(e)})

@app.route('/api/captain/form-based-analyzer', methods=['GET'])
def api_form_based_captain_analyzer():
    """API endpoint for Form-based Captain Analyzer"""
    try:
        results = form_based_captain_analyzer()
        return jsonify({"status": "ok", "players": results})
    except Exception as e:
        return jsonify({"status": "error", "message": str(e)})

@app.route('/api/captain/matchup-based-advisor', methods=['GET'])
def api_matchup_based_captain_advisor():
    """API endpoint for Matchup-based Captain Advisor"""
    try:
        results = matchup_based_captain_advisor()
        return jsonify({"status": "ok", "players": results})
    except Exception as e:
        return jsonify({"status": "error", "message": str(e)})

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5001, debug=True)

File: archive/server-node-backup/server/captain-api.ts
import express from "express";
import { exec } from "child_process";

const captainApi = express.Router();

function runTool(tool, res) {
  const cmd = `python3 -c "from captain_tools import ${tool}; import json; print(json.dumps(${tool}()))"`;
  exec(cmd, (err, stdout) => {
    if (err) return res.status(500).json({ status: "error", message: err.message });
    try {
      const data = JSON.parse(stdout);
      res.json({ status: "ok", ...data });
    } catch (error) {
      res.status(500).json({ status: "error", message: "Invalid JSON from Python" });
    }
  });
}

captainApi.get("/score-predictor", (_, res) => runTool("captain_score_predictor", res));
captainApi.get("/vice-captain-optimizer", (_, res) => runTool("vice_captain_optimizer", res));
captainApi.get("/loophole-detector", (_, res) => runTool("loophole_detector", res));
captainApi.get("/form-based-analyzer", (_, res) => runTool("form_based_captain_analyzer", res));
captainApi.get("/matchup-based-advisor", (_, res) => runTool("matchup_based_captain_advisor", res));

export default captainApi;